src/foam/lib/json/Outputter.java

```diff
@@ -96,17 +97,33 @@ protected void outputNull() {
   }
 
   public void outputString(String s) {
-    writer_.append("\"");
-    writer_.append(escape(s));
-    writer_.append("\"");
+    if ( multiLineOutput_ && s.indexOf('\n') >= 0 ) {
+      writer_.append("\n  \"\"\"\n    ");
+      s = ltrim(s, 4);
+      writer_.append(escapeMultiline(s));
+      writer_.append("\n  \"\"\"");
+    }
+    else {
+      writer_.append("\"");
+      writer_.append(escape(s));
+      writer_.append("\"");
+    }
   }
 
   public String escape(String s) {
-    return s
-      .replace("\\", "\\\\")
-      .replace("\"", "\\\"")
-      .replace("\t", "\\t")
-      .replace("\n","\\n");
+    return s.replace("\\", "\\\\")
+            .replace("\"", "\\\"")
+            .replace("\t", "\\t")
+            .replace("\n","\\n");
+  }
+
+  public String escapeMultiline(String s) {
+    s = s.replace("\\", "\\\\")
+         .replace("\\\"", "\"")
+         .replace("\n","\\n")
```

Comment:
Don't unescape newlines, we're in a multiline output so newlines are okay

Reply:
this is just to add indentation formatting when we run s.replace("\\n", "\n    "); that way all new lines are indented

Reply:
Are you sure?  If you escape newlines to be \\n then later if you do .replace("\n", ...) it won't match anything since the newline is now escaped.